README.md

```diff
@@ -1384,6 +1386,56 @@ out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade s
    }
 }
 ```
+### Virtual meter service
+
+This service enables a device to report accumulated consumption in case it does not have its own metering capabilities (eg. a relay with output binary switch service or thermostat). Thanks to the service accumulated consumption is calculated based on manually provided consumption in a unit of time (eg. watts, m3/h). A device shall calculate cumulated consumption at every state's change, mode change and at least every interval value - 30 minutes by default. When this service is added on a device, appropriate metering service is created (eg. meter_elec, meter_water) and a device starts to send measurements.
+
+#### Service names
+
+`meter_virtual`
+
+#### Interfaces
+
+Type | Interface                | Value type |   Unit  | Description
+-----|--------------------------|------------|---------|------------
+in   | cmd.set_interval         | int        | minutes | Interval  for accumulated consumption recalculation. Overwrites a default value.
+in   | cmd.add                  | int_float  | define by `unit` props   | Adds Virtual meter service to a selected device to report accumulated consumption. Map of floats shall provide consumption for every mode.
+in   | cmd.remove               | null       |         | Removes Virtual meter service from a selected device. The device shall not be reporting accumulated consumption.
```

Comment:
Command to remove acts like "remove all". Thing if you want to have fine-grained behavior where you can remove consumption set for a single unit and remove all at once. I'd suggest having two interfaces, one for removing a particular unit and one for removing all.

Reply:
Good point but I am wondering. Virtual meter is added per MQTT address which denotes "%d_%d" (devID, endpoinID). We do not support more then 1 virtual meter per this address. So isn't it unambiguous if we send cmd.remove for any %d_%d address ?

Reply:
@boguslaw-wojcik ?

Reply:
@BudzikM Negative, this is not how this behavior should in my opinion be done.

Imagine that you support for a virtual electricity three units: W, A, V. 
* User configures wattage consumption and sends add command for W unit.
* Adapter adds for the first time meter_elec service and declares it supports W and kWh in relevant attributes, then sends updated inclusion report.
* User configures amperage consumption and sends add command for A unit.
* Adapter adds to the already created meter_elec service another supported unit A and sends updated inclusion report.
* User configures voltage and sends add command for V unit.
* Adapter adds to the already created meter_elec service another supported unit V and sends updated inclusion report.

Conclusions:
* You should not create multiple meter_elec services
* Adapter must have a logic based on units it allows to virtualize, understand if they are pulse or cumulative, how to spoof them and calculate, and to which service they should be added.

As for the implementation, currently I think you may just implement it as supporting just one unit (W) and do not hassle about all the complexities, but the design of API should allow such extension in future.

Reply:
This is some misconoception partially, we should not base any action on unit. Unit is a unit. It depends on a device what capabilities it has and unit just need to match it. V doesn make any phisical sense, what would be the result of voltage * delta_t ?. Mutliple meters will not be supported on the same device, action is to overwrite the previous one. If a device is kind of meter elec and it supports energy metering, sending any unit different that Watts should return error and device makes this decision.

Reply:
Let's have a call.

Reply:
Had discussion in a call. Resolved.